core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -107,7 +107,11 @@ public String tableName() {
   }
 
   public TableMetadata startMetadata() {
-    return current;
+    return this.base;
```

Comment:
`startMetadata()` wasn't used before and I was expecting it to point to `this.base`